iceoryx_hoofs/include/iceoryx_hoofs/cxx/helplets.hpp

```diff
@@ -155,8 +155,15 @@ struct range
 template <typename T>
 T align(const T value, const T alignment) noexcept
 {
-    T remainder = value % alignment;
-    return value + ((remainder == 0U) ? 0U : alignment - remainder);
+    T remainder{value % alignment};
```

Comment:
It is almost certainly correct and there is a simple proof for this. Negation leads to correct bitmasking so that `&` acts as modulo for powers of two. 

We should not check restrictions (as runtime check) but document them. The restrictions are that alignment has to be a power of two and `v + a - 1 <= max value` of the type (this is the worst case alignment that can happen).

Reply:
Some explanation. If `a` is a power of two, say 2^k, then exactly the kth bit is set in `a`. `-a` creates a bit mask where only the lowest k-1 bits are set. Meaning masking any number `v` with it by `&` always gives a multiple of `a`.

Now the observation is that if `v` is already a multiple of `a`, then `v + a -1` is smaller than the next multiple of `a` , that is `v+a` and masking produces `v` (as it zeros the lowest bits). 

Otherwise `v + a` is between the next multiple of `a` (call it `m` ) but smaller than the one after it `m+a`. Hence masking returns `m`.

@FerdinandSpitzschnueffler  @saif-at-github

Reply:
@saif-at-github Could you try replacing the function implementation as follows

```
/// @note value + alignment - 1 must not exceed the maximum value for type T
/// @note alignment must be a power of two
T align(const T value, const T alignment) noexcept
{
   return (value + alignment - 1) & (-alignment);
}
```
If all tests are green with this, we should keep it like this (the formula is correct, I sketched the proof above). The constraint in the note is required in any case. Otherwise we have a test where alignment is not a power of two (this test would be wrong, it always is a power of two as mandated by the standard).

Reply:
@MatthiasKillat corrected. all tests are green with this.

Reply:
@saif-at-github Can you please add the Doxygen notes as well?

Reply:
If there is another change we should add the documentation I mentioned above.